tests/e2e/multi_vc_preferential_topology.go

```diff
@@ -0,0 +1,776 @@
+		nimbusGeneratedK8sVmPwd     string
+		allowedTopologies           []v1.TopologySelectorLabelRequirement
+		ClusterdatastoreListVC      []map[string]string
+		ClusterdatastoreListVC1     map[string]string
+		ClusterdatastoreListVC2     map[string]string
+		ClusterdatastoreListVC3     map[string]string
+		parallelStatefulSetCreation bool
+		stsReplicas                 int32
+		scaleDownReplicaCount       int32
+		scaleUpReplicaCount         int32
+		stsScaleUp                  bool
+		stsScaleDown                bool
+		verifyTopologyAffinity      bool
+		topValStartIndex            int
+		topValEndIndex              int
+		topkeyStartIndex            int
+		scParameters                map[string]string
+		storagePolicyInVc1Vc2       string
+		allowedTopologyLen          int
+		parallelPodPolicy           bool
+		nodeAffinityToSet           bool
+		podAntiAffinityToSet        bool
+		snapc                       *snapclient.Clientset
+		pandoraSyncWaitTime         int
+		err                         error
+	)
+
+	ginkgo.BeforeEach(func() {
+		var cancel context.CancelFunc
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+
+		multiVCbootstrap()
+
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, defaultNginxStorageClassName, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name,
+				*metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		nimbusGeneratedK8sVmPwd = GetAndExpectStringEnvVar(nimbusK8sVmPwd)
+		sshClientConfig = &ssh.ClientConfig{
+			User: "root",
+			Auth: []ssh.AuthMethod{
+				ssh.Password(nimbusGeneratedK8sVmPwd),
+			},
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		}
+
+		stsScaleUp = true
+		stsScaleDown = true
+		verifyTopologyAffinity = true
+		scParameters = make(map[string]string)
+		storagePolicyInVc1Vc2 = GetAndExpectStringEnvVar(envStoragePolicyNameInVC1VC2)
+		topologyMap := GetAndExpectStringEnvVar(topologyMap)
+		allowedTopologies = createAllowedTopolgies(topologyMap, topologyLength)
+
+		//Get snapshot client using the rest config
+		restConfig = getRestConfigClient()
+		snapc, err = snapclient.NewForConfig(restConfig)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// fetching k8s master ip
+		allMasterIps = getK8sMasterIPs(ctx, client)
+		masterIp = allMasterIps[0]
+
+		// fetching cluster details
+		clientIndex := 0
+		clusterComputeResource, _, err = getClusterNameForMultiVC(ctx, &multiVCe2eVSphere, clientIndex)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// fetching list of datastores available in different VCs
+		ClusterdatastoreListVC1, ClusterdatastoreListVC2,
```

Comment:
Please use camelcase in all the variables.

Reply:
Ack